acra/acra-in-depth/data-structures/_index.md

```diff
@@ -2,3 +2,180 @@
+
+Which mode should you choose?
+
+Let’s consider an example where we’re storing an email in a database and we’d like to encrypt it, “wrapping” it into an CryptoEnvelope. We’d get a table:
+
+| Email        | Column 2     | Column 3     |
+|--------------|--------------|--------------|
+| <CryptoEnvelope> | Column2Value | Column3Value |
+| <CryptoEnvelope> | Column2Value | Column3Value |
+
+In this case, an CryptoEnvelope takes up a whole cell and we are trying to decrypt it as is, without searching for anything.
+But things can be different. In InjectedCell mode a binary [MsgPack](https://msgpack.org/index.html) or [protobuf](https://developers.google.com/protocol-buffers/) could be stored in a table, and it is possible that in those data entities only one field is encrypted.
+
+Such data entity wouldn’t be a single CryptoEnvelope - it would be a data entity that contains an CryptoEnvelope or several CryptoEnvelopes. This means that in the InjectedCell mode we stop assuming that the database cells can only contain complete CryptoEnvelopes. CryptoEnvelopes can be inside some other pieces of data and that’s where we’re starting to look for them. This consequently slows down the processing speed. However, not every task needs the InjectedCell mode, which is why one can switch between modes, depending on what you’re encrypting and how you’re storing it.
+
+By default, the WholeCell mode is active (`--acrastruct_wholecell_enable`). You can switch between the modes using startup parameters:
+
+```
+--acrastruct_injectedcell_enable
+    AcraStruct may be injected into any place of data cell
+--acrastruct_wholecell_enable
+    AcraStruct will stored in whole data cell (default true)
+```
+
+Both these flags switch mode for AcraBlocks too.
+
+Also see the [CLI reference]({{< ref "acra/configuring-maintaining/general-configuration/acra-server.md#acraserver-cli-reference-INVALID" >}}).
+
+## Zone Ids
+
+Zone Ids are identifiers of EC keys used for matching Zones to keys. The current format is 8 bytes "begin tag" + 16 symbols a-zA-Z. For example: `DDDDDDDDzxzXVyBBaNclkgPS` where `DDDDDDDD` is "begin tag" and `zxzXVyBBaNclkgPS` is unique key identifier. 
+
+## When use AcraBlocks and when AcraStructs
+
+If you need end-to-end encryption then you should choose AcraStructs and encrypt data on client side using public key. In all other cases you should prefer AcraBlocks that have less output size, fewer keys size and are more performant because use only symmetric key encryption. When AcraStructs rely on asymmetric plus symmetric encryption algorithms and are slower. AcraServer supports transparent AcraBlock/AcraStruct encryption and decryption with equal functionality.
```

Comment:
IMO, this should be somewhere closer to the top since this is something people would want to read. Making a choice between AcraBlocks and AcraStructs is important for the users. Implementation details and internal structure is something a user does not care much about, that's not their problem.